linkedList.js

```diff
@@ -0,0 +1,372 @@
+class Node {
+  constructor(data) {
+    this._data = data;
+    this.nextNode = null;
+  }
+
+  getData() {
+    return this._data;
+  }
+
+
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  length() {
+    var currentNode = this.head;
+    var length = 0;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode) {
+      length++;
+      currentNode = currentNode.nextNode;
+    }
+    return length;
+  }
+
+  insert(value) {
+    var node = new Node(value);
+    var currentNode = this.head;
+
+    if(!currentNode) {
+      this.head = node;
+
+      return node;
+    }
+
+    currentNode = this.head;
+    node.nextNode = currentNode;
+    this.head = node;
+
+    this._length++;
+    return node;
+  }
+
+  search(value) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return false;
+    }
+
+    while(currentNode.nextNode) {
+      if(currentNode.getData() === value) {
+        return true;
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return false;
+  }
+
+  findMax() {
+    var currentNode = this.head;
+    var max = this.head.getData();
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() > max) {
+        max = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return max;
+  }
+
+  findMin() {
+    var currentNode = this.head;
+    var min = this.head.getData();
+
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() < min) {
+        min = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return min;
+  }
+
+  findNthFromBeginning(index) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+    for(var i = 0; i < index; i++) {
+      currentNode = currentNode.nextNode;
+    }
+
+    return (currentNode ? currentNode.getData() : null);
+  }
+
+  visit() {
+    var printString = "";
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      printString += currentNode.getData() + " ";
+      currentNode = currentNode.nextNode;
+    }
+
+    console.log(printString);
+  }
+
+  insertAscending(value) {
+    var currentNode = this.head;
+    var node = new Node(value);
+    var inserted = false;
+
+
+
+    if(!currentNode) {
+      return null;
```

Comment:
If currentNode is null, i.e. the head is null, then you'd want to update head to be the new node and not just return null.